job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/JenkinsJobManagement.java

```diff
@@ -215,10 +218,15 @@ public String createOrUpdateConfigFile(ConfigFile configFile, boolean ignoreExis
             );
         }
 
-        configProvider.save(config);
+        // save all config files on global scope - config-file-provider would support folders too
+        getGlobalConfigFileStore().save(config);
```

Comment:
Arg. This does not work. `ConfigProvider#save(Config)` did store the configs in a map, mapping ID to config. So saving a config with an existing ID would replace the old config. Now, with `GlobalConfigFileStore` it creates a second config with an identical ID since `GlobalConfigFileStore` does not ensure unique IDs.